tests/AeroSharp.UnitTests/DataAccess/Validation/MapConfigurationValidatorTests.cs

```diff
@@ -0,0 +1,35 @@
+﻿using AeroSharp.DataAccess;
+using AeroSharp.DataAccess.Validation;
+using FluentAssertions;
+using NUnit.Framework;
+using System.Collections;
+
+namespace AeroSharp.UnitTests.DataAccess.Validation;
+
+[TestFixture]
+internal sealed class MapConfigurationValidatorTests
+{
+    private static readonly MapConfigurationValidator Validator = new();
+
+    [Test]
+    [TestCaseSource(nameof(TestCases))]
+    public void Validate_MapConfiguration_ReturnsExpectedResult(MapConfiguration mapConfiguration, bool expectedIsValid)
+    {
+        // act
+        var result = Validator.Validate(mapConfiguration);
+
+        // assert
+        result.IsValid.Should().Be(expectedIsValid);
+    }
+
+    private static IEnumerable TestCases
+    {
+        get
+        {
+            yield return new TestCaseData(new MapConfiguration { CreateOnly = true, UpdateOnly = true }, false);
+            yield return new TestCaseData(new MapConfiguration { CreateOnly = true, UpdateOnly = false }, true);
+            yield return new TestCaseData(new MapConfiguration { CreateOnly = false, UpdateOnly = true }, true);
+            yield return new TestCaseData(new MapConfiguration { CreateOnly = false, UpdateOnly = false }, true);
```

Comment:
Is it valid to be neither?

Reply:
Yeah, in the case when neither are set, the [default policy](https://docs.aerospike.com/apidocs/csharp/html/t_aerospike_client_mapwriteflags) is used, which allows both create and update.